packages/pendle/src/Pendle.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  type TransactionFilter,
+  type SwapActionParams,
+  compressJson,
+} from '@rabbitholegg/questdk'
+import { type Address } from 'viem'
+import { SUPPORTED_CHAINS_ARRAY, type SupportedChainId } from './chain-ids'
+import { getRouterAddress } from './contract-addresses'
+import { SwapYTV3_ABI } from './abi'
+
+export const swap = async (
+  swap: SwapActionParams,
+): Promise<TransactionFilter> => {
+  const { chainId } = swap
```

Comment:
As it stands the only parameters you'll be able to create quests for would be chain ID, I'd recommend adding a token and token amount parameter as well.

Reply:
Hi, currently we are allowing all tokens, allow selecting specific token in or tokens out would require too much handling, for example:
- If the token in is normal ERC20 tokens, then the token out must be either YT or PT. 
- If the token in is YT, tokens out are either normal ERC20 tokens or PT
- if the token in is PT, tokens out are either normal ERC20 tokens or YT
- ... same for tokens out
Not to mention that we have multiple PT/YT tokens.